Documentation/ExtensionArchitecture/Tutorials/Tea/Controller.rst

```diff
@@ -0,0 +1,56 @@
+..  include:: /Includes.rst.txt
+
+..  index::
+    Tutorial Tea; Controller.rst
+..  _extbase_tutorial_tea_controller:
+
+==========
+Controller
+==========
+
+The controller controls the flow of data between the view and the
+data repository containing the model.
+
+A controller can contain one or more actions. Each of them is a method which
+ends on the name "Action" and returns an object of type
+:php:`\Psr\Http\Message\ResponseInterface`.
+
+In the following action a tea object should be displayed in the view:
+
+.. include:: /CodeSnippets/Tutorials/Tea/Classes/Domain/Controller/ShowAction.rst.txt
+
+This action would be displayed if an URL like the following would be requested:
+:samp:`https://www.example.org/myfrontendplugin?tx_tea[action]=show&tx_tea[controller]=tea&tx_tea[tea]=42&chash=whatever`.
+
+So where does the model :php:`Tea $tea` come from? The only reference we had
+to the actual tea to be displayed was the id :php:`42`. In most cases the
+parent class :php:`ActionController` will take care of matching parameters to
+objects or models. In more advanced scenarios it is necessary to influence
+the parameter matching. But in our scenario it is sufficient to know this
+happens automatically in the controller.
+
+The following action expects not parameters. It fetches all available tea
+objects from the repository and hands them to the view:
+
+..  include:: /CodeSnippets/Tutorials/Tea/Classes/Domain/Controller/IndexAction.rst.txt
+
+The controller has to access the :php:`TeaRepository` to find all available tea
+objects. We use :ref:`Dependency Injection <DependencyInjection>` to make the
+repository available to the controller: The method :php:`injectTeaRepository()`
```

Comment:
The method "injectTeaRepository" is not available in the snippet above.

Reply:
updated the snippet